### DISCUSSION TITLE
X-Forwarded-For

### DISCUSSION THREAD

**Discussion by chrisla23** [Security-Onion-Solutions/securityonion#6754] [Unsupported Versions] [answered]
Is it possible to disable an Alert based on the contents of an X-Forwarded-For: header IP from network.data.decoded?

Ideally to disable the alert altogether, failing that have the IP from the X-Forwarded-For: header broken out into a field (not necessarily the one for the source IP) to be sorted/excluded in the UI? 

About all, I have been able to find on this is an old discussion in the google groups mentioning there may be some functionality to replace the source IP with the xff IP but that it may break correlation with pcap data. 

I am also not clear how that would be configured in the modern version should that choice be acceptable.  I apologize if I am missing it as searching including the word forward on the forum takes me to many unrelated posts. 

I am already blocking these false alerts with BPF based on source IP,  but these are slipping through since the client machine is making the request on the scanner's behalf. 

Thanks,

-Chris

**Comment by weslambert** [comment, marked as answer]
You might try modifying your Suricata config to contain the following for `xff`:

```
xff:
  enabled: yes
  mode: overwrite 
```

I believe it should replace the source IP with what is contained in the XFF header.  At that point, you could then threshold/suppress as you wish.

See: https://www.furkancaliskan.com/replacing-source-ip-with-x-forwarded-for-in-suricata/

**Reply by chrisla23** [reply]
I will give that a go. It would be a helpful improvement if either the default or an example yaml file had every possible valid definition defined. 

Thanks!

-Chris
